dns_check/tests/test_dns_check.py

```diff
@@ -0,0 +1,154 @@
+from .mocks import MockTime, success_query_mock, nxdomain_query_mock
+# stdlib
+import mock
+
+# 3p
+from dns.resolver import Resolver, Timeout
+
+# project
+from datadog_checks.dns_check import DNSCheck
+
+RESULTS_TIMEOUT = 10
+
+CONFIG_SUCCESS = {
+    'instances': [{
+        'name': 'success',
+        'hostname': 'www.example.org',
+        'nameserver': '127.0.0.1'
+    }, {
+        'name': 'cname',
+        'hostname': 'www.example.org',
+        'nameserver': '127.0.0.1',
+        'record_type': 'CNAME'
+    }]
+}
+
+CONFIG_SUCCESS_NXDOMAIN = {
+    'name': 'nxdomain',
+    'hostname': 'www.example.org',
+    'nameserver': '127.0.0.1',
+    'record_type': 'NXDOMAIN'
+}
+
+CONFIG_DEFAULT_TIMEOUT = {
+    'init_config': {
+        'default_timeout': 0.1
+    },
+    'instances': [{
+        'name': 'default_timeout',
+        'hostname': 'www.example.org',
+        'nameserver': '127.0.0.1'
+    }]
+}
+
+CONFIG_INSTANCE_TIMEOUT = {
+    'name': 'instance_timeout',
+    'hostname': 'www.example.org',
+    'timeout': 0.1,
+    'nameserver': '127.0.0.1'
+}
+
+CONFIG_INVALID = [
+    # invalid hostname
+    ({'name': 'invalid_hostname',
+        'hostname': 'example'}, "DNS resolution of example has failed"),
+    # invalid nameserver
+    ({'name': 'invalid_nameserver',
+        'hostname': 'www.example.org',
+        'nameserver': '0.0.0.0'}, "DNS resolution of www.example.org timed out"),
+    # invalid record type
+    ({'name': 'invalid_rcrd_type',
+        'hostname': 'www.example.org',
+        'record_type': 'FOO'}, "DNS resolution of www.example.org has failed"),
+    # valid domain when NXDOMAIN is expected
+    ({'name': 'valid_domain_for_nxdomain_type',
+        'hostname': 'example.com',
+        'record_type': 'NXDOMAIN'}, "DNS resolution of example.com has failed"),
+]
+
+
+@mock.patch('datadog_checks.dns_check.dns_check.time_func', side_effect=MockTime.time)
+@mock.patch.object(Resolver, 'query', side_effect=success_query_mock)
+def test_success(mocked_query, mocked_time, aggregator):
+    integration = DNSCheck('dns_check', {}, {})
+
+    integration.check(CONFIG_SUCCESS['instances'][0])
+    tags = ['instance:success', 'resolved_hostname:www.example.org', 'nameserver:127.0.0.1', 'record_type:A']
+    aggregator.assert_service_check(DNSCheck.SERVICE_CHECK_NAME, status=DNSCheck.OK,
+                                    tags=tags, count=1)
+
+    aggregator.assert_metric('dns.response_time', tags=tags, count=1, value=1)
+
+    integration.check(CONFIG_SUCCESS['instances'][1])
+    tags = ['instance:cname', 'resolved_hostname:www.example.org', 'nameserver:127.0.0.1', 'record_type:CNAME']
+    aggregator.assert_service_check(DNSCheck.SERVICE_CHECK_NAME, status=DNSCheck.OK,
+                                    tags=tags, count=1)
+    aggregator.assert_metric('dns.response_time', tags=tags, count=1, value=1)
+
+    # Assert coverage for this check on this instance
+    aggregator.assert_all_metrics_covered()
+    aggregator.reset()
```

Comment:
You can remove the manual resets since the fixture you put does that automatically.

Reply:
Good catch - Thanks